Directory.Build.targets

```diff
@@ -1,11 +1,21 @@
 <Project>
-  <!-- Taken from Microsoft.VisualStudio.SDK.EmbedInteropTypes NuGet package which tries to fix this issue
-       (but doesn't include this assembly in their list).  -->
-  <Target Name="LinkVSEmbeddableAssemblies" DependsOnTargets="ResolveReferences" AfterTargets="ResolveReferences">
+
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
+    <PackageReference Include="Microsoft.VisualStudio.SDK.EmbedInteropTypes" Version="15.0.15" />
+  </ItemGroup>
+
+  <!-- Don't embed interop types from Microsoft.VisualStudio.Setup.Configuration.Interop (which the
+       Microsoft.VisualStudio.SDK.EmbedInteropTypes package enables by default), because otherwise
+       VtblGap methods get generated in the reference source, resulting in compilation errors:
+          CS0531: 'ISetupInstance2._VtblGap1_8()': interface members cannot have a definition
+    -->
+  <Target Name="LinkVSEmbeddableAssemblies" DependsOnTargets="ResolveReferences" AfterTargets="LinkVSSDKEmbeddableAssemblies"
+          BeforeTargets="FindReferenceAssembliesForReferences">
     <ItemGroup>
       <ReferencePath Condition="'%(FileName)' == 'Microsoft.VisualStudio.Setup.Configuration.Interop'">
-        <EmbedInteropTypes>true</EmbedInteropTypes>
+        <EmbedInteropTypes>false</EmbedInteropTypes>
```

Comment:
Doesn't this mean we have to redistribute them?

Reply:
Yeah this is wrong. Adding that package above should set `EmbedInteropTypes` to `true` which is what we want (passes them as `/link` rather than a reference to the compiler).

Reply:
I'm not sure, I think they're in the GAC.  I had problems with the reference assembly source when this was enabled.  If we have to embed interop types for this assembly we'll have to find a way to fix the reference assembly issue.